common/api/imodeljs-frontend.api.md

```diff
@@ -7480,20 +7500,54 @@ export function readElementGraphics(bytes: Uint8Array, iModel: IModelConnection,
 // @internal
 export function readPointCloudTileContent(stream: ByteStream, iModel: IModelConnection, modelId: Id64String, _is3d: boolean, range: ElementAlignedBox3d, system: RenderSystem): RenderGraphic | undefined;
 
+// @alpha
+export class RealityDataConnectionManager {
+    // (undocumented)
+    getFromSourceKey(rdSourceKey: RealityDataSourceKey, iTwinId: GuidString | undefined): Promise<IRealityDataConnection | undefined>;
+    // (undocumented)
+    static get instance(): RealityDataConnectionManager;

```

Comment:
Eww singleton?

Reply:
"Manager" objects and singletons tend to be code smells. What is the singleton - or the class itself - buying you over just having a function to achieve the same thing?

I'm more interested in why you feel it is necessary to have an unbounded cache of IRealityDataConnection. This function is only used when creating a new tile tree. The cache seems unnecessary. I propose defining this function as `IRealityDataConnection.fromSourceKey` and removing the manager and the cache.

Reply:
In previous implementation, the metadata extracted from ContextShare (aka RealityDataService) is cached somewhere along the implementation of the RealityModelTileTree and need to be fetch again when reality data is detach and then re-attached from/to display style. 
E.g.: In Reality Data Web Viewer, in the 4D navigation, we can do this operation several time when we only display a reality data for a selected date. We list in the reality data widget all reality data available for the project, and they are attached/detached from display style when user change the date and time he want to see. 
This code is my tentative to an optimization to decouple the lifecycle of the RealityData metadata from the actual display system. 
I've created the manager to actually managed that persistance outside the display system.
I do see workflow where we will probably want to do similar for display tiles. Currenlty, the tiles are flushed and downloaded back when a reality data is detached from view and then reattach. This is far from ideal for performance.
I don't think we should remove the cache, at least not in 2.19. 
In 3.0 this caching mechanism should probably be moved into RealityDataAccess Implementation (something I'll need to look at a little more closely).
But I can add a static member to RealityDataConnection to keep the cache and do what you propose: RealityDataConnection.fromSourceKey.

Reply:
Sounds good.

Reply:
Push new fix that use a static method instead of creating a singleton class to obtain a RealityDataConnection instance.

Reply:
Thanks. I pushed another change to remove the "manager" entirely.